osf/models/subject.py

```diff
@@ -29,8 +36,32 @@ def child_count(self):
     def get_absolute_url(self):
         return self.absolute_api_v2_url
 
-    @property
+    @cached_property
     def hierarchy(self):
-        if self.parents.exists():
-            return self.parents.first().hierarchy + [self._id]
+        if self.parent:
+            return self.parent.hierarchy + [self._id]
         return [self._id]
+
+    @cached_property
+    def object_hierarchy(self):
+        if self.parent:
+            return self.parent.object_hierarchy + [self]
+        return [self]
+
+    @classmethod
+    def create(cls, text, provider, parent=None, bepress_subject=None):
```

Comment:
This seems to only add validation. Can we move this validation around so that we don't need to use this special method? I'm concerned that a developer would accidentally bypass validation by constructing an object the normal way.